src/components/Main.js

```diff
@@ -0,0 +1,74 @@
+import { $ } from '../util/querySelector';
+import Movie from './Movie';
+import MovieSkeleton from './MovieSkeleton';
+
+const SKELETON_ITEM_COUNT = 20;
+
+class Main {
+  #element;
+  #manager;
+
+  constructor (element, manager) {
+    this.#element = element;
+    this.#manager = manager;
+
+    this.#requestMovieListEvent();
+
+    this.#element.innerHTML = `
+    <h2></h2>
+    <ul class="item-list"></ul>
+    `;
+  }
+
+  renderSkeleton () {
+    const query = this.#manager.getQuery();
+    const skeleton = new MovieSkeleton();
+
+    $('h2', this.#element).innerHTML = query === '' ? '지금 인기 있는 영화' : `"${query}" 검색 결과`;
+    $('ul', this.#element).appendChild(document.createElement('div'));
+
+    const skeletonElement = $('ul', this.#element).lastElementChild;
+    skeletonElement.outerHTML = skeleton.template().repeat(SKELETON_ITEM_COUNT);
+  }
+
+  async render () {
+    const movie = new Movie();
+    const query = this.#manager.getQuery();
+
+    if (query === '' && !this.#manager.getMovieList().length) {
+      this.renderSkeleton();
+      await this.#manager.searchMovieList('');
+    } else if (this.#manager.getCurrentPage() === 1) {
+      this.renderSkeleton();
+      await this.#manager.searchMovieList(query);
+    }
+
+    this.#element.innerHTML = `
+    <h2>${query === '' ? '지금 인기 있는 영화' : `"${query}" 검색 결과`}</h2>
+    <ul class="item-list">
+    ${
+      this.#manager.getMovieList().length
+      ? this.#manager.getMovieList().map((movieInfo) => movie.template(movieInfo)).join('\n')
+      : '<p>검색 결과가 없습니다.</p>'
+    }
+    </ul>
+    ${
+      this.#manager.isLastPage()
+      ? ''
+      : '<button class="btn primary full-width">더 보기</button>'
+    }
```

Comment:
```suggestion
    ${
      !this.#manager.isLastPage() ?? 
      '<button class="btn primary full-width">더 보기</button>'
    }
```
빈값을 주는 조건이 있다면 이런식의 표현은 어떨까요?

추가로 ?? || 등을 같이 공부해보면 좋을거 같네요~

Reply:
Logical OR assignment과 Nullish coalescing operator도 찾아볼게요 감사합니다!!

하지만 `??`는 null이나 undefiened일 경우 후자를 반환하는데 위 경우에는 `isLastPage()`가 boolean을 반환해서 동작하지 않습니다. `||` 역시 아래와 같이 사용한다면 버튼을 출력해야 할 때는 잘 출력하지만, 마지막 페이지에 도달했을 경우  버튼 대신 'true'라는 문자열이 반환되는 형식이라 추가적인 처리가 더 필요해서 사용하기는 불편할 것 같습니다. 
```suggestion
    ${
      this.#manager.isLastPage() || 
      '<button class="btn primary full-width">더 보기</button>'
    }
```

Reply:
isLastPage()가 boolean을 반환하군요 ㅠㅠ 제가 코드를 잘못 본거 같아요
저도 고민해봤는데 그럼 말씀하신대로 참새가 처음 해준 방식이 제일 좋겠네요 👍